hcloud/load_balancers.go

```diff
@@ -87,6 +91,57 @@ func (l *loadBalancers) GetLoadBalancerName(ctx context.Context, clusterName str
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+// MustGetKey creates a key from Kubernetes resource in the format <namespace>/<name>
+func MustGetKey(obj interface{}) string {
+	if obj == nil {
+		return ""
+	}
+
+	o := MustGetObject(obj)
+	return fmt.Sprintf("%s/%s", o.GetNamespace(), o.GetName())
+
+}
+
+// MustGetObject casts the object into a Kubernetes `metav1.Object`
+func MustGetObject(obj interface{}) metav1.Object {
+	if obj == nil {
+		return nil
+	}
+
+	if oma, ok := obj.(metav1.ObjectMetaAccessor); ok {
+		return oma.GetObjectMeta()
+	} else if o, ok := obj.(metav1.Object); ok {
+		return o
+	}
+
+	panic(fmt.Errorf("Unknown type: %v", reflect.TypeOf(obj)))
```

Comment:
Calling panic runs the risk of the cloud controller manager crashing or ending up in an undefined state. It is better to log the error and return a sensible default, e.g. `nil`, instead of panicing. Returning `nil` does have other consequences, which I address in other comments.